hat/menupermissions/apps.py

```diff
@@ -1,5 +1,58 @@
 from django.apps import AppConfig
+from django.db.models.signals import post_migrate
+from hat.menupermissions.constants import MODULE_PERMISSIONS, MODULES
+
+
+def get_module(permission_code_name, iaso_permission, iaso_module):
+    for module_codename, permissions in MODULE_PERMISSIONS.items():  # loop on modules from constants
+        if permission_code_name in permissions:  # check if the permission is in the current module
+            iaso_permission = iaso_permission.objects.filter(
+                module__codename=module_codename, permission__codename=permission_code_name
+            )
+            if iaso_permission:  # check if the link between the current module and the permission exists
+                return None
+            else:
+                module = iaso_module.objects.filter(codename=module_codename)
+                if module:  # check if the module already exists in the db
+                    return module.first()
+                else:
+                    filtered_module = list(
+                        filter(lambda module: module["codename"] == module_codename, MODULES)
+                    )  # get the module to be created
+                    return iaso_module.objects.create(
+                        name=filtered_module[0]["name"], codename=filtered_module[0]["codename"]
+                    )
+
+
+def create_iaso_permissions_callback(sender, **kwargs):
+    from iaso.models import Permission as IasoPermission, Module
+    from django.contrib.auth.models import Permission as AuthPermission
+
+    if "plan" in kwargs:
+        plan = kwargs["plan"]
+        if plan:  # Check if there is a new migration
+            (plan_migration, plan_bool) = plan[0]
+            if not plan_bool:  # Check if it's not a rollback migration
+                operation = plan_migration.operations[0]
```

Comment:
I had this fail on a migration that was just a merge, which means, I think that there are no operations in it. It made me realize that this is probably called on ALL migrations. Is that what was intended?

Reply:
Yes my intention was to run this code every time a new hat.menupermissions is added.